nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/asset/StandardConnectorAssetManager.java

```diff
@@ -0,0 +1,225 @@
+
+        try {
+            Files.copy(contents, assetFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        } catch (final Exception e) {
+            throw new IOException("Failed to write connector asset to file " + assetFile.getAbsolutePath(), e);
+        }
+
+        final String digest = computeDigest(assetFile);
+        final Asset asset = new StandardAsset(assetId, connectorId, assetName, assetFile, digest);
+        assets.put(assetId, asset);
+        return asset;
+    }
+
+    private void deleteFile(final File file) {
+        try {
+            Files.delete(file.toPath());
+        } catch (final IOException e) {
+            logger.warn("Failed to delete [{}]", file.getAbsolutePath(), e);
+        }
+    }
+
+    private String generateAssetId() {
+        return UUID.randomUUID().toString();
+    }
+
+    private File getFile(final String connectorId, final String assetId, final String assetName) {
+        final Path parentPath = assetStorageLocation.toPath().normalize();
+        final Path assetPath = Paths.get(connectorId, assetId, assetName).normalize();
+        final Path fullPath = parentPath.resolve(assetPath);
+        return fullPath.toFile();
+    }
+
+    private String getStorageLocation(final AssetManagerInitializationContext initializationContext) {
+        final String storageLocation = initializationContext.getProperties().get(ASSET_STORAGE_LOCATION_PROPERTY);
+        return storageLocation == null ? DEFAULT_ASSET_STORAGE_LOCATION : storageLocation;
+    }
+
+    private void recoverLocalAssets() throws IOException {
+        final File[] connectorDirectories = assetStorageLocation.listFiles();
+        if (connectorDirectories == null) {
+            throw new IOException("Unable to list files for connector asset storage location %s".formatted(assetStorageLocation.getAbsolutePath()));
+        }
+
+        for (final File connectorDirectory : connectorDirectories) {
+            if (!connectorDirectory.isDirectory()) {
+                continue;
+            }
+
+            final String connectorId = connectorDirectory.getName();
+            final File[] assetIdDirectories = connectorDirectory.listFiles();
+            if (assetIdDirectories == null) {
+                logger.warn("Unable to list directory [{}]", connectorDirectory.getAbsolutePath());
+                continue;
+            }
+
+            for (final File assetIdDirectory : assetIdDirectories) {
+                final String assetId = assetIdDirectory.getName();
+
+                final File[] assetFiles = assetIdDirectory.listFiles();
+                if (assetFiles == null) {
+                    logger.warn("Unable to list directory [{}]", assetIdDirectory.getAbsolutePath());
+                    continue;
+                }
+
+                for (final File assetFile : assetFiles) {
+                    final String assetName = assetFile.getName();
+                    final String digest = computeDigest(assetFile);
```

Comment:
If we expect a bunch of assets, especially large ones, it may also be worth considering adding the digest to the filename so that it's not necessary to compute on startup every time. Again, not necessary here but worth considering for a future PR.